### PR TITLE
Feat/omit warnings

### LINKED ISSUE #25: Analyze & fix thrown warnings/errors in datapreparation
When running datapreparation / cleaning methods, there are warnings and errors thrown. We should analyze what's wrong there and whether the exists an error or whether these are just hints/warnings. If these are hints/warnings, we want to deny the logging of them.

### PR DESCRIPTION
**Pull request by MaxStroh**
fixed thrown errors/warnings and deleted pandas supress options and todos

### CODE REVIEW

File: nextbike/datapreparation/data_clean.py [hunk 1]
```diff
@@ -22,8 +19,8 @@ def data_cleaning(p_df_original):
     # Drop Duplicates and creating new df with only unique files
     p_df_original.drop_duplicates(subset=p_df_original.columns.difference(["p_lng", "p_lat"]), inplace=True)
     # Drop trip first/last
-    df_clean_unique_trip = p_df_original[(p_df_original["trip"] == "start") | (p_df_original["trip"] == "end")]
-    df_clean_unique_trip.sort_values(["b_number", "datetime"], inplace=True)
+    df_clean_unique_trip_unsorted = p_df_original[(p_df_original["trip"] == "start") | (p_df_original["trip"] == "end")]
+    df_clean_unique_trip = df_clean_unique_trip_unsorted.sort_values(["b_number", "datetime"])
```
Comment (weisske) [inline nextbike/datapreparation/data_clean.py]: I think we should do it consistently... 
Therefore, replace all inplace=True by override of df or creation of ne df.

**Review by Speedfan94 — APPROVED** [review]
Approved

File: nextbike/io/input.py [hunk 2]
```diff
@@ -18,7 +18,10 @@ def read_csv(p_filename, p_io_folder, p_sub_folder=""):
     """
     path = os.path.join(get_data_path(), p_io_folder, p_sub_folder, p_filename)
     try:
-        df = pd.read_csv(path, index_col=0)
+        # index_col=0 throws numpy warning,
+        # so we set it manually after reading in the file
+        df = pd.read_csv(path)
+        df.set_index(df.columns[0], inplace=True)
```
Comment (weisske) [inline nextbike/io/input.py]: Holy shit I allways searched were this warning came from :D
nice :+1:

**Review by weisske — CHANGES_REQUESTED** [review]
Changing inplace=True

File: nextbike/datapreparation/data_clean.py [hunk 3]
```diff
@@ -64,11 +61,11 @@ def data_cleaning(p_df_original):
     # Only keep trips, which where merged correctly
     df_merged = df_merged[df_merged["b_number_start"] == df_merged["b_number_end"]]
 
-    df_merged.drop(
+    df_merged = df_merged.drop(
         ["key_0",
-         "b_number_end"], axis=1, inplace=True
+         "b_number_end"], axis=1
```
Comment (weisske) [inline nextbike/datapreparation/data_clean.py:66]: reindexing earlier instead of drop key_0 would be better

**Review by weisske — APPROVED** [review]
Approve but change of drop["key_0"] is worse than reindexing

**Review by mulko93 — APPROVED** [review]
cool implementation!

File: nextbike/datapreparation/data_clean.py (hunk 1 above)
Comment (MaxStroh) [reply nextbike/datapreparation/data_clean.py]: replaced all inplace options in last commit :)

File: nextbike/io/input.py (hunk 2 above)
Comment (MaxStroh) [reply nextbike/io/input.py]: 🥇 :D

File: nextbike/datapreparation/data_clean.py (hunk 3 above)
Comment (MaxStroh) [reply nextbike/datapreparation/data_clean.py:66]: I just removed the inplace=True option here :D
I thought key_0 is something produced by the merge? How should we drop this before merging then? 🤔

**Pull request merged**